### PR TITLE
Andrew

### PR DESCRIPTION
**Pull request by AndrewAKG**

### CODE REVIEW

File: app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import org.ieeeguc.ieeeguc.models.User.Type;
 import org.json.JSONObject;
 
+import java.io.Console;
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java]: why is this import here

Comment (AndrewAKG) [reply app/src/main/java/org/ieeeguc/ieeeguc/controllers/LoginActivity.java]: by mistake i think....hshelo (y)

File: app/src/main/java/org/ieeeguc/ieeeguc/fragments/user/UserUpdate.java [hunk 2]
```diff
@@ -2,23 +2,87 @@
...
+    public void notifyUser(String message){
+        Snackbar.make(getActivity().findViewById(android.R.id.content), message,
+                Snackbar.LENGTH_INDEFINITE).setAction("Ok", new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+            }
+        }).show();
     }
+    public void updateProfile() {
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/fragments/user/UserUpdate.java]: document your method

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 3]
```diff
@@ -16,6 +18,8 @@
 import okhttp3.RequestBody;
 import okhttp3.Response;
 
+import static org.ieeeguc.ieeeguc.controllers.MainActivity.token;
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: weird import

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 4]
```diff
@@ -1,5 +1,7 @@
 package org.ieeeguc.ieeeguc.models;
 
+import android.widget.TextView;
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: weird import

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 5]
```diff
@@ -187,6 +191,100 @@ public void onResponse(Call call, Response response) throws IOException {
     }
 
     /**
+     * this method is called when a user wants to get the info of a specific committee
+     * @param {String} id [id of the desired Committee]
+     * @param {HTTPResponse} HTTP_RESPONSE [HTTPResponse interface instance]
+     * @return {void}
+     */
+    public static void getCommittee(String id,final HTTPResponse HTTP_RESPONSE){
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: remove the `getCommittee` function from here

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 6]
```diff
@@ -187,6 +191,100 @@ public void onResponse(Call call, Response response) throws IOException {
...
+                        HTTP_RESPONSE.onFailure(x,json);
+                    }
+                } catch (Exception e) {
+                    HTTP_RESPONSE.onFailure(500,null);
+                }
+                response.close();
+            }
+        });
+        }catch(JSONException e){
+            HTTP_RESPONSE.onFailure(-1,null);
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: change the status code to `500`

File: .idea/inspectionProfiles/Project_Default.xml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<component name="InspectionProjectProfileManager">
```
Comment (KEMDAK) [inline .idea/inspectionProfiles/Project_Default.xml]: revise this file

File: .idea/inspectionProfiles/profiles_settings.xml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<component name="InspectionProjectProfileManager">
```
Comment (KEMDAK) [inline .idea/inspectionProfiles/profiles_settings.xml]: revise this file

**Review by KEMDAK — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/ieeeguc/ieeeguc/controllers/MainActivity.java [hunk 9]
```diff
@@ -38,11 +36,25 @@ protected void onCreate(Bundle savedInstanceState) {
 
         // setting the context
         context = this;
+/*
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/controllers/MainActivity.java]: remove the testing

File: app/src/main/java/org/ieeeguc/ieeeguc/fragments/user/UserUpdate.java [hunk 10]
```diff
@@ -6,19 +6,94 @@
...
+        if (!(oldPassword.isEmpty()) && phoneNumber1.matches("\\+?\\d\\d\\d-?\\d\\d\\d-?\\d\\d\\d?\\d?\\d?\\d?")) {
+            MainActivity.loggedInUser.editProfile(MainActivity.token, oldPassword, newPassword, memberShipID, phoneNumber1, new HTTPResponse() {
+                public void onSuccess(int statusCode, JSONObject body) {
+                    MainActivity.createSnackBar("profile updated Successfully");
+                }
+
+                public void onFailure(int statusCode, JSONObject body) {
+                    if(statusCode == 400){
+                        try {
+                            MainActivity.createSnackBar(body.getString("message"));
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/fragments/user/UserUpdate.java:72]: the handling of the 400 request is not like this!!

**The message is for you as a developer not for the end user**

you will find in the body and array called `error` it contains the validation errors each one with the field and its problem.

**Leave it for now we will talk about it in the retrospective meeting**

**Review by KEMDAK — CHANGES_REQUESTED** [review]
well done. keep it up :D

**Review by KEMDAK — APPROVED** [review]

**Pull request merged**